tools/slicec-cs/src/decoding.rs

```diff
@@ -397,6 +415,42 @@ fn decode_stream_parameter(type_ref: &TypeRef, namespace: &str, encoding: Encodi
     }
 }
 
+fn decode_result_field(type_ref: &TypeRef, namespace: &str, encoding: Encoding) -> CodeBlock {
+    if type_ref.is_optional {
+        let mut decode_func_body = decode_func_body(type_ref, namespace, encoding);
+
+        // TODO: it's lame to have to do this here. We should provide a better API.
+        if matches!(type_ref.concrete_type(), Types::Sequence(_) | Types::Dictionary(_)) {
```

Comment:
What would be the better API you'd expect? In psuedo-code.

Reply:
Easy: we should provide a Rust function that creates a CodeBlock with the full decoding logic, including this cast. Not repeat this cast code n times.

We even have different variations of this code, with a cast and with an `as` (like here).